Add doc to repo (#37)
stolostron/multicluster-role-assignment: README.md

# Multicluster Role Assignment Operator

[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)
[![Go Report Card](https://goreportcard.com/badge/github.com/stolostron/multicluster-role-assignment)](https://goreportcard.com/report/github.com/stolostron/multicluster-role-assignment)

> The Multicluster Role Assignment Operator is a Kubernetes operator that manages role-based access control (RBAC) across multiple clusters in a multicluster environment. It provides a declarative way to assign cluster roles to users, groups, or service accounts across multiple managed clusters.

Reviewer: ```suggestion
The Multicluster Role Assignment Operator is a Kubernetes operator that manages role-based access control (RBAC) across multiple clusters in a multicluster environment. It provides a declarative way to assign cluster roles to users and groups across multiple managed clusters.
```

#### ClusterSelection Fields

| Field | Type | Description | Required |
|-------|------|-------------|----------|
| `type` | `string` | Type of cluster selection (currently only `clusterNames`) | Yes |
| `clusterNames` | `[]string` | List of cluster names to target | Yes |

#### Status

The operator reports the status of each role assignment, including:
- Overall conditions for the `MulticlusterRoleAssignment`
- Individual status for each role assignment (Pending, Active, Error)
- Detailed messages for troubleshooting

## Examples

### Multiple Role Assignments

```yaml
apiVersion: rbac.open-cluster-management.io/v1alpha1
kind: MulticlusterRoleAssignment
metadata:
  name: admin-multi-role
spec:
  subject:
    kind: User
    name: admin-user
    apiGroup: rbac.authorization.k8s.io
  roleAssignments:
  - name: prod-admin
    clusterRole: admin
    targetNamespaces:
    - production
    clusterSelection:
      type: clusterNames
      clusterNames:
      - prod-cluster-1
      - prod-cluster-2
  - name: dev-edit
    clusterRole: edit
    targetNamespaces:
    - development
    clusterSelection:
      type: clusterNames
      clusterNames:
      - dev-cluster-1
```

> 
> ### Service Account Assignment
> 
> ```yaml
> apiVersion: rbac.open-cluster-management.io/v1alpha1
> kind: MulticlusterRoleAssignment
> metadata:
>   name: monitoring-sa
> spec:
>   subject:
>     kind: ServiceAccount
>     name: monitoring-agent
>     namespace: monitoring
>   roleAssignments:
>   - name: metrics-reader
>     clusterRole: system:monitoring
>     clusterSelection:
>       type: clusterNames
>       clusterNames:
>       - cluster-1
>       - cluster-2
>       - cluster-3
> ```

Reviewer: ```suggestion
```
